dashboard/app/views/devise/shared/_oauth_links.haml

```diff
@@ -33,3 +35,22 @@
   if (!!window.MakerBridge) {
     $('.with-google_oauth2').prop({'disabled': true, 'title': 'Please use the ‘Log in with Google’ button in the toolbar at the top of the Maker App. If you do not see a ‘Log in with Google’ button, please download the latest Maker App version.'});
   }
+
+  function select_google_oauth2_login() {logOAuthSignUpTypeSelected('google');}
+  function select_microsoft_v2_auth_login() {logOAuthSignUpTypeSelected('microsoft');}
+  function select_facebook_login() {logOAuthSignUpTypeSelected('facebook');}
+
+  function logOAuthSignUpTypeSelected(provider) {
+    // Only log if user is in the sign-up flow (not sign-in)
+    if (window.location.href.includes('/sign_up')) {
+      const selectOauthLoginTypeEvent = new CustomEvent('customLogEvent', {
+        detail: {
+          eventName: 'SIGN_UP_LOGIN_TYPE_PICKED_EVENT',
+          properties: {
+            'user login type': provider
+          }
+        }
+      });
+      document.dispatchEvent(selectOauthLoginTypeEvent);
+    }
+  }
```

Comment:
same request here -- please extract existing js into a new js file, wrapping existing js in a `$(document).ready(` block. then add your new js code to that file.